src/cprover/solver_progress.cpp

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************\
+
+Module: Solver Progress Reporting
+
+Author: Daniel Kroening, dkr@amazon.com
+
+\*******************************************************************/
+
+/// \file
+/// Solver Progress Reporting
+
+#include "solver_progress.h"
+
+#include "console.h"
+
+#include <iostream>
+
+void solver_progresst::operator()(size_t current)
+{
+  if(verbose)
+  {
+  }
+  else
+  {
+    if(first)
+    {
+      first = false;
+    }
+    else
+    {
+      if(consolet::is_terminal())
+        std::cout << "\x1b[1A\x1b[0K"; // up one row and clear the line
+    }
+
+    std::cout << consolet::orange << "Processing property " << (current + 1) << '/' << total
+              << consolet::reset << '\n';
+  }
+}
+
+void solver_progresst::finished()
+{
+  if(verbose)
+  {
+  }
+  else
+  {
+    if(consolet::is_terminal())
+    {
```

Comment:
What is the reason for not doing `if(!verbose && consolet::is_terminal())` here?

Reply:
I'll eventually get around to adding sensible output for the `--verbose` case!